packages/core/crates/core-strategy/src/strategy.rs

```diff
@@ -71,18 +70,26 @@ impl Display for MultiStrategy {
 
 #[async_trait(? Send)]
 impl SingularStrategy for MultiStrategy {
-    async fn on_tick(&self) -> Result<()> {
+    async fn on_tick(&self) -> Result<(Vec<ChannelEntry>, Vec<(Address, Balance)>)> {
+        let mut to_close: Vec<ChannelEntry> = vec![];
+        let mut to_open: Vec<(Address, Balance)> = vec![];
         for strategy in self.strategies.iter() {
-            if let Err(e) = strategy.on_tick().await {
-                error!("error on_tick in strategy {strategy}: {e}");
-
-                if !self.cfg.on_fail_continue {
-                    warn!("{self} on_tick chain stopped at {strategy}");
-                    return Err(e);
-                }
+            match strategy.on_tick().await {
+                Ok((to_close_from_singular_strategy, to_open_from_singular_strategy)) => {
+                    to_close = [to_close, to_close_from_singular_strategy].concat();
+                    to_open = [to_open, to_open_from_singular_strategy].concat();
+                },
+                Err(e) => {
+                    error!("error on_tick in strategy {strategy}: {e}");
+
+                    if !self.cfg.on_fail_continue {
+                        warn!("{self} on_tick chain stopped at {strategy}");
+                        return Err(e);
+                    }
+                },
             }
         }
-        Ok(())
+        Ok((to_close, to_open))
```

Comment:
Not needed, why not return `Result<()>`

Reply:
Changed back and introduced `collect_tick_decision` method in Promiscuous strategy to get the intermediate result